rpc/rpc-transport/socket/src/name.c

```diff
@@ -116,8 +116,17 @@ af_unix_client_bind(rpc_transport_t *this, struct sockaddr *sockaddr,
         }
 
         addr = (struct sockaddr_un *)sockaddr;
-        strcpy(addr->sun_path, path);
-        ret = bind(sock, (struct sockaddr *)addr, sockaddr_len);
+
+        int count = 0;
```

Comment:
This code is fine. However, if you could re-factor this function slightly and place this condition before line 109 along with line 110, it could help to remove duplication of line 111 and gf_log at 112.

Reply:
Yeah sure.

Reply:
I think you don't need to run a loop and count characters to coverity bug. You can directly use strncpy instead of using strcpy and pass the length sizeof(addr->sun_path) like below
strncpy(addr->sun_path, path, sizeof(addr->sun_path));
addr->sun_path[sizeof(addr->sun_path)-1] = '\0';

Reply:
I was thinking like this from 108 to 142 - 
        path_data = dict_get_sizen(this->options, "transport.socket.bind-path");
        char *path = data_to_str(path_data);
        if (path) {
		while (path[count] != '\0') {
		    count++;
		}
	}
   	if (!count && count <=108) {
	    addr = (struct sockaddr_un *)sockaddr;
            strcpy(addr->sun_path, path);
            ret = bind(sock, (struct sockaddr *)addr, sockaddr_len);
	    if (ret == -1) {
            gf_log(this->name, GF_LOG_ERROR,
                   "cannot bind to unix-domain socket %d (%s)", sock,
                   strerror(errno));
            goto err;
           }
    	} else {
        	gf_log(this->name, GF_LOG_TRACE,
               "bind-path not specified for unix socket, "
               "letting connect to assign default value");
    	}

Reply:
> I think you don't need to run a loop and count characters to coverity bug. You can directly use strncpy instead of using strcpy and pass the length sizeof(addr->sun_path) like below
> strncpy(addr->sun_path, path, sizeof(addr->sun_path));
> addr->sun_path[sizeof(addr->sun_path)-1] = '\0';

Yeah that would be better.